src/backend/marsha/core/tests/api/video/test_create.py

```diff
@@ -1,1106 +1,522 @@
-"""Tests for the Video API of the Marsha project."""
-from datetime import datetime, timedelta
-import json
+"""Tests for the Video create API of the Marsha project."""
+from datetime import timedelta
 import random
-from unittest import mock
 import uuid
 
-from django.conf import settings
-from django.test import TestCase, override_settings
-
-from boto3.exceptions import Boto3Error
+from django.test import TestCase
 
+from marsha.core import factories, models
+from marsha.core.api import timezone
+from marsha.core.defaults import INITIALIZED, STATE_CHOICES
 from marsha.core.simple_jwt.factories import (
-    InstructorOrAdminLtiTokenFactory,
     PlaylistLtiTokenFactory,
     StudentLtiTokenFactory,
     UserAccessTokenFactory,
 )
 
-from .. import api, factories, models
-from ..api import timezone
-from ..defaults import (
-    APPROVAL,
-    CC_BY,
-    CC_BY_SA,
-    ENDED,
-    HARVESTED,
-    HARVESTING,
-    IDLE,
-    INITIALIZED,
-    JITSI,
-    JOIN_MODE_CHOICES,
-    LIVE_CHOICES,
-    PENDING,
-    RAW,
-    READY,
-    RUNNING,
-    STARTING,
-    STATE_CHOICES,
-    STOPPED,
-    STOPPING,
-)
-from ..factories import LiveSessionFactory, VideoFactory
-from ..utils.api_utils import generate_hash
-from ..utils.medialive_utils import ManifestMissingException
-from ..utils.time_utils import to_timestamp
-from .utils import RSA_KEY_MOCK
-
-
-# We don't enforce arguments documentation in tests
-# pylint: disable=unused-argument,too-many-lines
 
-
-class VideoAPITest(TestCase):
-    """Test the API of the video object."""
+class VideoCreateAPITest(TestCase):
+    """Test the create API of the video object."""
 
     maxDiff = None
 
-    def test_api_video_read_detail_anonymous(self):
-        """Anonymous users should not be allowed to read a video detail."""
-        video = factories.VideoFactory()
-        response = self.client.get(f"/api/videos/{video.id}/")
+    def test_api_video_create_anonymous(self):
+        """Anonymous users should not be able to create a new video."""
+        response = self.client.post("/api/videos/")
```

Comment:
This will be unpleasant, but this commit breaks file history…
Maybe we should cleanup each test file through multiple commits, ensuring us that only deletions are applied ?

Reply:
Thanks for spotting that :+1: